commons/src/main/java/com/powsybl/commons/config/MapModuleConfig.java

```diff
@@ -126,74 +154,118 @@ public void setStringListProperty(String name, List<String> stringList) {
+    public Optional<Float> getOptionalFloatProperty(String name) {
+        Objects.requireNonNull(name);
+        Object value = properties.get(name);
+        if (value == null) {
+            return Optional.empty();
+        }
+        if (value instanceof Number) {
+            return Optional.of(((Number) value).floatValue());
+        } else if (value instanceof String) {
+            return Optional.of(Float.parseFloat((String) value));
+        } else {
+            throw createUnexpectedPropertyTypeException(value);
+        }
     }
 
     @Override
     public float getFloatProperty(String name) {
-        return Float.parseFloat(getStringProperty(name));
+        return getOptionalFloatProperty(name).orElseThrow(() -> createPropertyNotSetException(name));
     }
 
     @Override
     public float getFloatProperty(String name, float defaultValue) {
-        return Float.parseFloat(getStringProperty(name, Float.toString(defaultValue)));
+        return getOptionalFloatProperty(name).orElse(defaultValue);
     }
 
     @Override
-    public Optional<Float> getOptionalFloatProperty(String name) {
-        String value = getStringProperty(name, null);
-        return Optional.ofNullable(value != null ? Float.parseFloat(value) : null);
+    public OptionalDouble getOptionalDoubleProperty(String name) {
+        Objects.requireNonNull(name);
+        Object value = properties.get(name);
+        if (value == null) {
+            return OptionalDouble.empty();
+        }
+        if (value instanceof Number) {
+            return OptionalDouble.of(((Number) value).doubleValue());
+        } else if (value instanceof String) {
+            return OptionalDouble.of(Double.parseDouble((String) value));
+        } else {
+            throw createUnexpectedPropertyTypeException(value);
+        }
     }
 
     @Override
     public double getDoubleProperty(String name) {
-        return Double.parseDouble(getStringProperty(name));
+        return getOptionalDoubleProperty(name).orElseThrow(() -> createPropertyNotSetException(name));
     }
 
     @Override
     public double getDoubleProperty(String name, double defaultValue) {
-        return Double.parseDouble(getStringProperty(name, Double.toString(defaultValue)));
+        return getOptionalDoubleProperty(name).orElse(defaultValue);
     }
 
     @Override
-    public boolean getBooleanProperty(String name) {
-        return Boolean.parseBoolean(getStringProperty(name));
+    public Optional<Boolean> getOptionalBooleanProperty(String name) {
+        Objects.requireNonNull(name);
+        Object value = properties.get(name);
+        if (value == null) {
+            return Optional.empty();
+        }
+        if (value instanceof Boolean) {
+            return Optional.of((Boolean) value);
+        } else if (value instanceof String) {
+            return Optional.of(Boolean.parseBoolean((String) value));
```

Comment:
Maybe also here?!

Reply:
Boolean.parseBoolean cannot throw an exception, check is done internally by "true".equals()